docs/components/compact.md

```diff
@@ -9,37 +9,160 @@ menu: components
+
+* By _unique_, we mean that the set of labels in a Prometheus instance must be different from all other sets of labels of
+your Prometheus instances, so that the compactor will be able to group blocks by Prometheus instance.
+* By _persistent_, we mean that one Prometheus instance must keep the same labels if it restarts, so that the compactor will keep
+compacting blocks from an instance even when a Prometheus instance goes down for some time.
+
+Natively Prometheus does not store external labels anywhere. This is why external labels are added only on upload time to the `ThanosMeta` section
+of `meta.json` in each block.
+
+> **NOTE:** In default mode the state of two or more blocks having same external labels and overlapping in time is assumed as an unhealthy situation.
+Refer to [Overlap Issue Troubleshooting](../operating/troubleshooting.md#overlaps) for more info. This results in compactor [halting](#halting).
+
+### Vertical Compactions
+
+Thanos and Prometheus supports vertical compaction, so process of compacting multiple streams of blocks into one.
+
+In Prometheus, this can be triggered by setting hidden flag in Prometheus and putting additional TSDB blocks within Prometheus
+local directory. Extra blocks can overlap with existing ones. When Prometheus detects that situation it performs `vertical compaction`
+which compacts overlapping blocks into single one. This is mainly used for **backfilling** purposes.
+
+In Thanos, it works similarly, but on bigger scale and using external labels for grouping as explained in [Compaction section](#compaction).
+
+In both systems, series with the same labels are merged together. Merging samples is **naive**. It works by deduplicating samples within
+exactly the same timestamps. Otherwise samples are added in sorted by time order.
+
+> **NOTE:** Both Prometheus and Thanos default behaviour is to fail compaction if any overlapping blocks are spotted. (For Thanos, within the same external labels).
+
+#### Vertical Compaction Use Cases
+
+There can be few valid use cases for vertical compaction:
+
+* Races between multiple compactions, for example multiple compactors or between compactor and Prometheus compactions. While this will have extra
+computation overhead for Compactor it's safe to enable vertical compaction for this case.
+* Backfilling. If you want to add blocks of data to any stream where there is existing data already there for the time range, you will need enabled vertical compaction.
+* Offline deduplication of series. It's very common to have the same data replicated into multiple streams. We can distinguish two common series duplications, `one-to-one` and `realistic`:
+  * `one-to-one` duplication is when same series (series with the same labels from different blocks) for the same range have **exactly** the same samples: Same values and timestamps.
+This is very common while using [Receivers](../components/receive.md) with replication greater than 1.
```

Comment:
Maybe a quick explanation: "Because this will de-duplicate the identically replicated samples produced."